### DISCUSSION TITLE
TypeError: Starlette.on_event() missing 1 required positional argument: 'event_type'

### DISCUSSION THREAD

**Discussion by andresprogramacion123** [fastapi/fastapi#6110] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from config.database import engine, SQLModel, create_db_and_tables

app = FastAPI

@app.on_event('startup')
async def on_startup():
    create_db_and_tables()
```

### Description

When executing the uvicorn main:app --reload command, it returns me in the shell error:

TypeError: Starlette.on_event() missing 1 required positional argument: 'event_type'

It seems to be a problem with starlette but I'm not sure

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

fastapi 0.89.1

### Python Version

3.10.6

**Comment by ebottos94** [comment, marked as answer]
Hi @andresprogramacion123, app variable must be FastAPI instance. 

```python
from fastapi import FastAPI
from config.database import engine, SQLModel, create_db_and_tables

app = FastAPI()

@app.on_event('startup')
async def on_startup():
    create_db_and_tables()
```

**Comment by ebottos94** [comment]
@andresprogramacion123 If your problem is solved can you close the issue? Thank you.

**Comment by andresprogramacion123** [comment]
@ebottos94 It was difficult for me to see it although it was simple, thank you very much.
